fedmsg_migration_tools/bridges.py

```diff
@@ -41,126 +39,52 @@ def zmq_to_amqp(amqp_url, exchange, zmq_endpoints, topics):
-
-    def __init__(self, amqp_url, queue_name, bindings, publish_endpoint):
-        self.connection = Connection(amqp_url)
-        self.bindings = bindings
-        self.queues = [Queue(queue_name)]
-        context = zmq.Context.instance()
-        self.publish_endpoint = publish_endpoint
-        self.pub_socket = context.socket(zmq.PUB)
-        self.pub_socket.bind(publish_endpoint)
-        _log.info('Bound to %s for ZeroMQ publication', publish_endpoint)
-
-    def get_consumers(self, consumer_class, channel):
-        """
-        Get the list of Consumers this class uses.
-
-        This is part of the :class:`kombu.mixins.ConsumerMixin` API.
-
-        Args:
-            consumer_class (class): The class to use when creating consumers.
-            channel (kombu.Channel): Unused.
-        """
-        return [
-            consumer_class(self.queues, callbacks=[self.on_message], accept=['json']),
-        ]
-
-    def on_consume_ready(self, connection, channel, consumers, **kwargs):  # pragma: no cover
-        """
-        Implement the ConsumerMixin API.
-
-        Args:
-            connection (kombu.Connection): Unused.
-            channel (kombu.Channel): Unused.
-            consumers (list): List of :class:`kombu.Consumer`. Unused.
-        """
-        _log.info('AMQP consumer ready on the %r queues', self.queues)
-        for consumer in consumers:
-            for queue in consumer.queues:
-                for bind in self.bindings:
-                    queue.bind_to(
-                        exchange=bind['exchange'], routing_key=bind['routing_key'],
-                        arguments=bind['arguments'])
-                    _log.info('Binding "%s" to %r', queue.name, bind)
-        super(AmqpConsumer, self).on_consume_ready(connection, channel, consumers, **kwargs)
-
-    def on_consume_end(self, connection, channel):  # pragma: no cover
-        """
-        Implement the ConsumerMixin API.
-
-        Args:
-            connection (kombu.Connection): Unused.
-            channel (kombu.Channel): Unused.
-        """
-        _log.info('Successfully canceled the AMQP consumer')
-        super(AmqpConsumer, self).on_consume_end(connection, channel)
-
-    def on_message(self, body, message):
-        """
-        The callback for the Consumer, called when a message is received.
-
-        As this consumer must run outside the reactor thread (since it uses blocking APIs)
-        this simply uses the Twisted API to call the delivery service's message handler inside
-        the reactor thread.
+    # ZMQ
+    context = zmq.Context.instance()
+    pub_socket = context.socket(zmq.PUB)
+    pub_socket.bind(publish_endpoint)
+    _log.info('Bound to %s for ZeroMQ publication', publish_endpoint)
 
-        Args:
-            body (dict): The decoded message body.
-            message (kombu.Message): The Kombu message object.
-        """
+    # AMQP
+    def on_message(ch, method, properties, body):
```

Comment:
Is this right? I think the callback should just accept a ``Message`` positional argument.